### ISSUE TITLE
Undefined index: object

### ISSUE THREAD

**Issue by JackWH** [beyondcode/laravel-query-detector#12]
Hi,

When I try and load my application since installing this, I'm getting "Undefined index: object" on just about every page load.

This error is being triggered by `/home/vagrant/sites/mysite/vendor/beyondcode/laravel-query-detector/src/QueryDetector.php` on line `60`.

As a test, I did `dd($relation)` just before line 60. The output of the dump is below — as you can see there's no `object` index. Not sure if this is caused by a quirk in my application, or something in the package?

```
array:6 [
  "file" => "/home/vagrant/sites/mysite/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php"
  "line" => 550
  "function" => "noConstraints"
  "class" => "Illuminate\Database\Eloquent\Relations\Relation"
  "type" => "::"
  "args" => array:1 [...]
]
```

Thanks in advance,

Jack

**Comment by mpociot** [comment]
Can you show me the code that you run? How do you access your relations?
I'm using `debug_backtrace(DEBUG_BACKTRACE_PROVIDE_OBJECT, 50)` so, there should be an "object" key present. 

I also published a new version (0.2.1) that checks for the object existence - but if the object key is not present, the query detection won't work.

**Comment by JackWH** [comment]
I'm struggling a bit to pin it down as to where in the application in causing this. Looking at the debug backtrace, the last bit of my application code that triggers before this errors is contained within an Eloquent global scope class, that runs on every page load and is assigned to multiple models. Here's the relevant code within the class:

```
// Try to get the user via their native session
if (! $this->user) {
    $this->user = auth()->user();
}
```

Specifically the call to `auth()->user();` seems to hand everything else over to Laravel behind the scenes, before the query detector takes over and errors.

In fact, if I just try and call `$test = \Auth::user();` right at the top of this global scope class, query detector fails immediately before it gets to any of the logic within the rest of the class. 

I'm not sure if it's something weird with having a global scope being applied to every model load?
